prov/util/src/util_mem_monitor.c

```diff
@@ -246,9 +291,14 @@ int ofi_monitors_add_cache(struct ofi_mem_monitor **monitors,
 		}
 
 		if (dlist_empty(&monitor->list)) {
-			ret = monitor->start(monitor);
-			if (ret)
-				goto err;
+			pthread_mutex_lock(&mm_state_lock);
+			start_list[iface] = monitor;
+			assert(monitor->state != FI_MM_STATE_UNSPEC);
```

Comment:
This assert could be moved into ofi_monitors_update.

Reply:
Done